src/legacy/core_plugins/tile_map/public/shim/legacy_dependencies_plugin.ts

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import chrome from 'ui/chrome';
+import { CoreStart, Plugin } from 'kibana/public';
+import { initTileMapLegacyModule } from './tile_map_legacy_module';
+
+/** @internal */
+export interface LegacyDependenciesPluginSetup {
+  serviceSettings: any;
+  $injector: any;
+}
+
+export class LegacyDependenciesPlugin
+  implements Plugin<Promise<LegacyDependenciesPluginSetup>, void> {
+  public async setup() {
+    initTileMapLegacyModule();
+
+    const $injector = await chrome.dangerouslyGetActiveInjector();
+
+    return {
+      $injector,
+      // Settings for EMSClient.
+      // EMSClient, which currently lives in the tile_map vis,
```

Comment:
EMSClient will likely be moved into its own github project and imported as an npm artifact.
https://github.com/elastic/kibana-team/issues/120